be/test/storage/binlog_file_test.cpp

```diff
@@ -0,0 +1,402 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+
+#include "fs/fs_util.h"
+#include "gen_cpp/binlog.pb.h"
+#include "storage/binlog_file_reader.h"
+#include "storage/binlog_file_writer.h"
+#include "storage/binlog_util.h"
+#include "testutil/assert.h"
+
+namespace starrocks {
+
+struct TestLogEntryInfo {
+    LogEntryPB log_entry;
+    int64_t version;
+    int64_t start_seq_id;
+    int64_t end_seq_id;
+    FileIdPB* file_id;
+    int32_t start_row_id;
+    int32_t num_rows;
+    bool end_of_version;
+    int64_t timestamp;
+};
+
+class BinlogFileTest : public testing::Test {
```

Comment:
For UT
1. add tests to generate larger(>100MB)/multiple version(>1000) binlog with randomly generated content, and read multiple versions to verify content correctness.
2. add tests that begin without commit/abort , all multiple repeat begin/commit/abort, see the behavior is expected

Reply:
For 1, add `test_random_write_read`; for 2, add `test_random_begin_commit_abort`